server/odc-service/src/main/java/com/oceanbase/odc/service/session/ConnectConsoleService.java

```diff
@@ -341,6 +446,22 @@ public List<SqlExecuteResult> getAsyncResult(@NotNull String sessionId, String r
         }
     }
 
+    public AsyncExecuteResultResp getAsyncResultV2(@NotNull String sessionId, String requestId) {
+        PreConditions.validArgumentState(Objects.nonNull(requestId), ErrorCodes.SqlRegulationRuleBlocked, null, null);
+        ConnectionSession connectionSession = sessionService.nullSafeGet(sessionId);
+        AsyncExecuteContext<SqlExecuteResult> executeContext =
+                (AsyncExecuteContext) ConnectionSessionUtil.getExecuteContext(connectionSession, requestId);
+        if (executeContext.getFuture().isDone()) {
+            ConnectionSessionUtil.removeExecuteContext(connectionSession, requestId);
+            return new AsyncExecuteResultResp(SqlExecuteStatus.SUCCESS, executeContext);
```

Comment:
SqlExecuteStatus is designed to illustrate the status of sql execution, we may use a boolean flag to indicated whether the execution is done or cancelled?

Reply:
ok , refactor it into a boolean val named `finished`.